### DISCUSSION TITLE
docker 部署的milvus集群，minio存储问题

### DISCUSSION THREAD

**Discussion by sunmingqiaa** [milvus-io/milvus#24264] [Q&A and General discussion] [answered]
minio:
  address: localhost # Address of MinIO/S3
  port: 9000   # Port of MinIO/S3
  accessKeyID: minioadmin # accessKeyID of MinIO/S3
  secretAccessKey: minioadmin # MinIO/S3 encryption string
  useSSL: false # Access to MinIO/S3 with SSL
  bucketName: "a-bucket" # Bucket name in MinIO/S3
  rootPath: files # The root path where the message is stored in MinIO/S3

如果将minio挂载多个磁盘目录，一个bucket映射到多个目录，rootpath可以在多个目录下同时存在吗，或者是rootpath可以配置多个目录来对应同一个bucket在不同目录下的不同文件夹。如：
/data1/a-bucket/files1
/data2/a-bucket/files2
对于minio需要存储数据量较多时，一个目录空间不够，该如何设置的问题有点迷。

**Comment by yhmo** [comment, marked as answer]
minio.rootPath 目前只能为1个，并且不同的milvus集群必须使用不同的rootpath，以免相互冲突

**Reply by sunmingqiaa** [reply]
也就是说单个milvus集群存储数据量取决于单个目录大小（挂载的盘的大小），这样不是有很大的存储限制吗，如果想扩容minio存储不知道怎么操作

**Comment by cyf33** [comment]
> 也就是说单个milvus集群存储数据量取决于单个目录大小（挂载的盘的大小），这样不是有很大的存储限制吗，如果想扩容minio存储不知道怎么操作

我也遇到了同样的问题……所以有好的方式可以扩容minio吗？

**Reply by LoveEachDay** [reply]
minio 扩容很方便，可以增加pod 的磁盘数，也可以增加serverpool。
milvus 使用的是minio bucket 的一个prefix, 这个底层是多个minio 服务的多块磁盘，不会受单盘大小的限制。

如果你是用helm chart 部署，扩容可以设置minio.zones=2, 就可以增加一倍的存储空间。

**Reply by cyf33** [reply]
感谢！之前试图增加replicas失败，原来还有zone的概念
